src/common/libjob/result.c

```diff
@@ -114,15 +115,16 @@ static int job_result_handle_exception (json_t *res,
     json_t *note = NULL;
     json_t *exception_occurred = json_object_get (res, "exception_occurred");
 
-    if (!exception_occurred)
-        return -1;
+    assert (exception_occurred);
```

Comment:
The assertion looks a bit worrisome.  Rather than invite cogitation on whether that could ever possibly be unset, maybe it would better to just replace the `json_is_true()` with `json_is_true (json_object_get (res, "exception_occurred"))` since `json_is_true()` is documented to return false on a NULL arg.